src/main/java/com/jmb/springfactory/service/productionorder/ProductionOrderServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package com.jmb.springfactory.service.productionorder;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.jmb.springfactory.dao.GenericMySQLService;
+import com.jmb.springfactory.dao.productionorder.ProductionOrderMySQL;
+import com.jmb.springfactory.exceptions.NotFoundException;
+import com.jmb.springfactory.exceptions.ServiceLayerException;
+import com.jmb.springfactory.model.bo.BusinessObjectBase;
+import com.jmb.springfactory.model.dto.ProductionOrderDTO;
+import com.jmb.springfactory.model.dto.ProductionScheduleDto;
+import com.jmb.springfactory.model.entity.ProductionOrder;
+import com.jmb.springfactory.service.GenericServiceImpl;
+import static com.jmb.springfactory.service.UtilsService.*;
+import com.jmb.springfactory.service.productionschedule.ProductionScheduleService;
+
+@Service
+public class ProductionOrderServiceImpl extends GenericServiceImpl<ProductionOrder, ProductionOrderDTO,
+  BusinessObjectBase, Integer> implements ProductionOrderService {
+
+  @Autowired 
+  private ProductionOrderMySQL productionOrderMySQL;
+  
+  @Autowired
+  private ProductionScheduleService productionScheduleService;
+
+  @Override
+  public GenericMySQLService<ProductionOrder, Integer> genericDao() {
+    return productionOrderMySQL;
+  }
+
+  @Override
+  public Class<? extends ProductionOrder> getClazz() {
+    return ProductionOrder.class;
+  }
+
+  @Override
+  public Class<? extends ProductionOrderDTO> getDtoClazz() {
+    return ProductionOrderDTO.class;
+  }
+
+  @Override
+  public Class<? extends BusinessObjectBase> getBoClazz() {
+    return BusinessObjectBase.class;
+  }
+  
+  @Override
+  public List<ProductionOrderDTO> findAllBySchedule(Integer idSchedule) throws NotFoundException {
+    return Optional.ofNullable(productionScheduleService.findOne(idSchedule))
+        .map(ProductionScheduleDto::getOrders)
+        .orElseThrow(NotFoundException::new);
+  }
+
+  @Override
+  public ProductionOrderDTO findOneByIdInSchedule(Integer idSchedule, Integer idOrder) throws NotFoundException {
+    return findAllBySchedule(idSchedule).parallelStream()
+        .filter(order -> order.getId().equals(idOrder))
+        .findFirst()
+        .orElseThrow(NotFoundException::new);
+  }
+  
+  @Override
+  public ProductionOrderDTO save(final ProductionOrderDTO order, Integer idSchedule) throws ServiceLayerException, 
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)